UnicornEngine/include/unicorn/video/Renderer.hpp

```diff
@@ -56,42 +56,55 @@ class Renderer
     bool Init();
     void Deinit();
     void Render();
-    static void onWindowResized(GLFWwindow* window, int width, int height);
+    static void OnWindowResized(GLFWwindow* window, int width, int height);
     bool RecreateSwapChain();
 
 private:
     bool m_isInitialized;
     GLFWwindow* m_pWindow;
-
-    VkInstance m_vkInstance;
-    VkPhysicalDevice m_vkPhysicalDevice;
-    VkDevice m_vkLogicalDevice;
-    VkSwapchainKHR m_vkSwapChain;
-    VkQueue m_graphicsQueue;
-    VkQueue m_presentQueue;
-    VkSurfaceKHR m_vkWindowSurface;
-    VkFormat m_swapChainImageFormat;
-    VkExtent2D m_swapChainExtent;
-    VkPipelineLayout m_pipelineLayout;
-    VkPipeline m_graphicsPipeline;
-    VkRenderPass m_renderPass;
-    VkCommandPool m_commandPool;
-    VkSemaphore m_imageAvailableSemaphore;
-    VkSemaphore m_renderFinishedSemaphore;
+    vk::Instance m_vkInstance;
+    vk::PhysicalDevice m_vkPhysicalDevice;
+    vk::Device m_vkLogicalDevice;
+    vk::SwapchainKHR m_vkSwapChain;
+    vk::Queue m_graphicsQueue;
+    vk::Queue m_presentQueue;
+    vk::SurfaceKHR m_vkWindowSurface;
+    vk::Format m_swapChainImageFormat;
+    vk::Extent2D m_swapChainExtent;
+    vk::PipelineLayout m_pipelineLayout;
+    vk::Pipeline m_graphicsPipeline;
+    vk::RenderPass m_renderPass;
+    vk::CommandPool m_commandPool;
+    vk::Semaphore m_imageAvailableSemaphore;
+    vk::Semaphore m_renderFinishedSemaphore;
     std::vector<const char*> m_validationLayers;
     std::vector<const char*> m_deviceExtensions;
     VkDebugReportCallbackEXT m_vulkanCallback;
     std::string m_gpuName;
-    std::vector<VkImage> m_swapChainImages;
-    std::vector<VkImageView> m_swapChainImageViews;
-    std::vector<VkFramebuffer> m_swapChainFramebuffers;
-    std::vector<VkCommandBuffer> m_commandBuffers;
+    std::vector<vk::Image> m_swapChainImages;
+    std::vector<vk::ImageView> m_swapChainImageViews;
+    std::vector<vk::Framebuffer> m_swapChainFramebuffers;
+    std::vector<vk::CommandBuffer> m_commandBuffers;
 #ifdef NDEBUG
     static const bool s_enableValidationLayers = false;
 #else
     static const bool s_enableValidationLayers = true;
 #endif
 
+    void FreeInstance();
+    void FreeDebugCallback();
+    void FreeSurface();
+    void FreeLogicalDevice();
+    void FreeSwapChain();
+    void FreeImageViews();
+    void FreeRenderPass();
+    void FreeGraphicsPipeline();
+    void FreePipelineLayout();
```

Comment:
Should it have `Create` pair? It is a part of `CreateGraphicsPipeline()` right now